store/boltdb/boltdb.go

```diff
@@ -127,12 +193,19 @@ func (b *BoltDB) Put(key string, value []byte, opts *store.WriteOptions) error {
 
 //Delete the value for the given key.
 func (b *BoltDB) Delete(key string) error {
-	db := b.client
+	var (
+		db  *bolt.DB
+		err error
+	)
+	if db, err = b.getDBhandle(); err != nil {
+		return err
+	}
+	defer b.releaseDBhandle()
 
-	err := db.Update(func(tx *bolt.Tx) error {
+	err = db.Update(func(tx *bolt.Tx) error {
 		bucket := tx.Bucket(b.boltBucket)
 		if bucket == nil {
-			return (ErrBoltBucketNotFound)
+			return store.ErrKeyNotFound
```

Comment:
Feels wrong to me to return `store.ErrKeyNotFound` here as it is not quite the same error happening and it's confusing for debugging the root cause of an issue with `boltDB`. This should be something like `store.ErrLocalStoreAccessRefused` or something more explicit.

It's fine because usually you use this local driver aside from the distributed drivers (zk, consul, etcd) and the code logic and error checks are different in this case. Might conflict in the future if we add more local drivers but it will do for now.